### DISCUSSION TITLE
🎉WALC IS ON THE AUR!! AND I MADE A .DEB FILE TOO!! AND SURPRISINGLY, I DIDN'T FAIL!!🎉

### DISCUSSION THREAD

**Discussion by albakhtari** [cstayyab/WALC#197] [General]
Not sure how to get the .deb file to @Zzombiee2361 though :thinking:

**Comment by albakhtari** [comment]
And I forgot to mention that both, the .deb installer and the AUR package, isntall to the directory: `/opt/WALC` and create a desktop entry: `/usr/share/applications/walc.desktop`

And one last thing @Zzombiee2361, to make my life easier, could you kindly set the naming of beta version as follows:
```bash
xxx.beta.xxx
# not...
xxx-beta-xxx
```
as dashes aren't allowed in the PKGBUILD version variable.
Thank you very, very much :grin:

**Reply by albakhtari** [reply]
I'm feeling lonely now...<h1>**CAN SOMEONE REPLY TO THIS POST SO I DON'T FEEL LIKE I'VE WASTED ALL THAT TIME...PLEASE!**</h1>

**Comment by daffamumtaz2361** [comment]
## Whoo...! Congrats on the AUR! 🥳 🎉

I'm not familiar with arch, is the pkgbuild need to be pushed to this repo or is it just fine to put it only in AUR?

<sub>We've always been able to build .deb file from `electron-builder` though</sub>

**Reply by albakhtari** [reply]
The AUR PKGBUILD has its own git repo (on the AUR): https://aur.archlinux.org/walc.git
When there's an update, I update the PKGBUILD file then push it, the verification is through my computer's public ssh sum.

Regarding the .deb installer, if you can deal with it then I think it's better because you can test it properly. I can test it on my Manjaro (arch) machine using `dpkg`, but it's still better to install it on a debian system. However, a .deb is a must.

So if we exclude the AppImage, we would have official support for Debian based and Arch based systems.
The only one left is Fedora. Shall I build a .rpm package for WALC?

**Reply by albakhtari** [reply]
I did some reading, and apparently you can build the electron project into any Linux installer. If you want me to do it, then let me know, I just want to make sure that we have covered all Linux distros.

If you're busy, then I can start working on the chat by number feature again, but I'll need some help on where to place the `newChat()` function and how to call it. Otherwise, I managed to get the other parts working (the prompt, the context menu and the shortcut).

**Comment by daffamumtaz2361** [comment]
I honestly prefer distro agnostic method of distribution like AppImage and flatpak (snap too but that's proven to be very problematic), since testing and building for many distro is honestly a pain

Also updating the app installed from `deb` or `rpm` is cumbersome unless I set up a repository (though I'm not sure about aur)

**Reply by albakhtari** [reply]
Yes, that makes sense. I forgot that you were making a flatpack.
